shell/platform/common/cpp/text_input_model.cc

```diff
@@ -75,9 +75,11 @@ void TextInputModel::AddText(const std::u16string& text) {
   if (selection_base_ != selection_extent_) {
     DeleteSelected();
   }
-  selection_extent_ = text_.insert(selection_extent_, text.begin(), text.end());
-  selection_extent_ += text.length();
-  selection_base_ = selection_extent_;
+  auto cursor_pos = text_.begin() + selection_extent_;
+  cursor_pos = text_.insert(cursor_pos, text.begin(), text.end());
+  cursor_pos += text.length();
+  selection_base_ = cursor_pos - text_.begin();
+  selection_extent_ = selection_base_;
```

Comment:
I believe you could replace all of this with:
```
text_.insert(selection_extent_, text);
selection_extent_ += text.length();
selection_base_ = selection_extent_;
```

Reply:
Nice catch. You can indeed. I've got no idea why I reworked it this way instead of similar to how I reworked the erase changes elsewhere :/ Thanks!